crates/catalog/hms/src/catalog.rs

```diff
@@ -95,9 +106,156 @@ impl HmsCatalog {
             client: HmsClient(client),
         })
     }
+
+    /// Create and extract properties from `hive_metastore::hms::Database`.
+    pub fn properties_from_database(database: &Database) -> HashMap<String, String> {
+        let mut properties = HashMap::new();
+
+        if let Some(description) = &database.description {
+            properties.insert("comment".to_string(), description.to_string());
+        };
+
+        if let Some(location) = &database.location_uri {
+            properties.insert("location".to_string(), location.to_string());
+        };
+
+        if let Some(owner) = &database.owner_name {
+            properties.insert(HMS_DB_OWNER.to_string(), owner.to_string());
+        };
+
+        if let Some(owner_type) = &database.owner_type {
+            let value = match owner_type {
+                PrincipalType::User => "User",
+                PrincipalType::Group => "Group",
+                PrincipalType::Role => "Role",
+            };
+
+            properties.insert(HMS_DB_OWNER_TYPE.to_string(), value.to_string());
+        };
+
+        if let Some(params) = &database.parameters {
+            params.iter().for_each(|(k, v)| {
+                properties.insert(k.clone().into(), v.clone().into());
+            });
+        };
+
+        properties
+    }
+
+    /// Converts name and properties into `hive_metastore::hms::Database`
+    /// after validating the `namespace` and `owner-settings`.
+    pub fn convert_to_database(
```

Comment:
How about we expose only two methods to convert between `Database` and `Namespace`, rather than `HashMap<String, String>` so that they these two apis are more consistent. We can have a new type such as following:
```rust
pub struct HmsDatabase(Database);
```

Reply:
I really like the idea. 

However, how would you handle the situation where I need to get the Database from the hive_metastore via the client. Fetching the db and then wrapping it with the new type simply to call `fn to_namespace` seems weird - but maybe there is another way to do this more elegantly?

Perhaps, we can extend the HmsCatalog as well with the new type?
This way I can fetch the DB with the client and set the db on HmsCatalog and call the conversion fns?

```rust
pub struct HmsDatabase(Database);

pub struct HmsCatalog {
  config: HmsCatalogConfig,
  client: HmsClient,
  database: HmsDatabase,
}
```

Reply:
> Fetching the db and then wrapping it with the new type simply to call fn to_namespace seems weird - but maybe there is another way to do this more elegantly?

This is what I think and currently I don't have idea to make it more elegant. To move on my suggestion is that we make the conversion apis crate private, and not exposing them before we found a good approach, what do you think?

Reply:
...in the python and java implementation similiar functions exists simply as private / static helper functions inside the respective `Catalog` classes. I think it would be fine if we do this as well (at least for now).

However, I'd still change the properties_from_database function to return a `Namespace` instead of a HashMap<String, String> which makes way more sense, like you suggested.

where should those functions live? inside `utils` or as associated fns on the catalog itself?